addons/web/static/src/js/views/search_panel.js

```diff
@@ -58,13 +60,28 @@ function _processSearchPanelNode(node, fields) {
         if (section.type === 'category') {
             section.icon = section.icon || 'fa-folder';
             section.hierarchize = !!pyUtils.py_eval(attrs.hierarchize || '1');
+            hasCategoryWithCounters = hasCategoryWithCounters || section.enableCounters;
         } else if (section.type === 'filter') {
             section.domain = attrs.domain || '[]';
             section.groupBy = attrs.groupby;
             section.icon = section.icon || 'fa-filter';
+            hasFilterWithDomain = hasFilterWithDomain || section.domain !== "[]";
         }
         sections[sectionId] = section;
     });
+    if (hasCategoryWithCounters && hasFilterWithDomain) {
+        // If incompatibilities are found -> disables all category counters
+        for (const section of Object.values(sections)) {
+            if (section.type === "category") {
+                section.enableCounters = false;
+            }
+        }
+        // ... and triggers a warning
+        console.warn(
+            "Warning: categories with counters are incompatible with filters having a domain attribute.",
+            "All category counters have been disabled to avoid inconsistencies.",
+        );
+    }
```

Comment:
I would add a comment saying that it seems possible to overcome the difficulty but that it would need a serious rework
of the search panel